design/sales.md

```diff
@@ -20,24 +19,142 @@ or being negotiated.
+that matches the requested amount, duration, and price. The matching
+availability will be removed from the reservations module, so that it can't be
+sold twice. If an availability matches, but is larger than the requested
+storage, then the Sales module may decide to split the availability into a part
+that we can use for the request, and a remainder that can be sold separately.
+
+It then selects a slot from the request to fill, and starts downloading its
+content chunk by chunk. For each chunk that is successfully downloaded, a bit of
+reserved space in the Repo is released. The content is stored in the Repo with a
+time-to-live value that ensures that the content remains in the Repo until the
+request expires.
+
+Once the entire content is downloaded, the sales module will calculate a storage
+proof, and submit the proof on chain. If these steps are all successful, then
+this node has filled the slot. Once the other slots are filled by other nodes
+the request will start. The time-to-live value of the content should then be
+updated to match the duration of the storage request.
+
+    Marketplace          Sales              Reservations      Repo
+      |                    |                     |              |
+      | incoming request   |                     |              |
+      |------------------->| find reservation    |              |
+      |                    |-------------------->|              |
+      |                    | remove reservation  |              |
+      |                    |-------------------->|              |
+      |                    |                     |              |
+      |                    | store content                      |
+      |                    |----------------------------------->|
+      |                    | set time-to-live                   |
+      |                    |----------------------------------->|
+      |                    | release reserved space             |
+      |                    |----------------------------------->|
+      |       submit proof |                                    |
+      |<-------------------|                                    |
+      |                    |                                    |
+      .                    .                                    .
+      .                    .                                    .
+      | request started    |                                    |
+      |------------------->| update time-to-live                |
+      |                    |----------------------------------->|
+
+Ending a request
+----------------
+
+When a storage request comes to an end, then the content can be removed from the
+repo and the storage space can be made available for sale again. The same should
+happen when something went wrong in the process of selling storage.
+
+The time-to-live value should be removed from the content in the Repo, reserved
```

Comment:
Should the ttl initially be set to the request expiry, then updated once the request has started? After that, does the ttl need to be removed?

Reply:
The request can fail, and then we need to remove the ttl.